src/clj/com/climate/claypoole/impl.clj

```diff
@@ -63,7 +63,16 @@
                                              (concat
                                               (.getStackTrace cause)
                                               (.getStackTrace e))))
-           (throw cause)))))
+           (throw cause)))
+       (catch clojure.lang.Compiler$CompilerException e
```

Comment:
I surrounded with try catch the eval function and un-wrapped the exception there. 
It looks a bit more clean IMO then adding a param to a bunch of functions. 
See PR #63 . 
I also formatted those files so they are a bit noisy. 
Sorry about that.